pkg/netpol/eval/resources.go

```diff
@@ -256,6 +308,7 @@ func (pe *PolicyEngine) deletePod(p *corev1.Pod) error {
 	if podObj, ok := pe.podsMap[podName]; ok {
 		// delete relevant workload entries from cache if all pods per owner are deleted
 		pe.cache.deletePod(podObj, podName)
+		delete(pe.ownersToLabelsMap, podObj.Owner.Name)
```

Comment:
if there are remaining pods of this owner, then the representative pod should be replaced rather than deleted.